src/service/viewer-impl.js

```diff
@@ -426,6 +432,10 @@ export class Viewer {
       this.lastVisibleTime_ = now;
       this.hasBeenVisible_ = true;
       this.whenFirstVisibleResolve_();
+      if (this.nextVisibleResolve_) {
+        this.nextVisibleResolve_();
+        this.nextVisibleResolve_ = null;
```

Comment:
uhmm shouldn't `nextVisiblePromise_`  also be nullified?

Reply:
ahhh! good catch